src/annotator/util/buckets.js

```diff
@@ -5,36 +5,43 @@ import { getBoundingClientRect } from '../highlighter';
  */
 
 /**
- * A tuple representing either the top (`startOrEnd` = 1) or the
- * bottom (`startOrEnd` = -1) of an anchor's highlight bounding box.
- *
- * @typedef {[pixelPosition: number, startOrEnd: (-1 | 1), anchor: Anchor]} PositionPoint
+ * @typedef Bucket
+ * @prop {Anchor[]} anchors - The anchors in this bucket
+ * @prop {number} position - The vertical pixel offset where this bucket should
+ *                           appear in the bucket bar.
  */
 
 /**
- * An object containing information about anchor highlight positions
- *
- * @typedef PositionPoints
- * @prop {Anchor[]} above - Anchors that are offscreen above
- * @prop {Anchor[]} below - Anchors that are offscreen below
- * @prop {PositionPoint[]} points - PositionPoints for on-screen anchor
- *   highlights. Each highlight box has 2 PositionPoints (one for top edge
- *   and one for bottom edge).
+ * @typedef WorkingBucket
+ * @prop {Anchor[]} anchors - The anchors in this bucket
+ * @prop {number} position - The computed position (offset) for this bucket,
+ *   based on the current anchors. This is centered between `top` and `bottom`
+ * @prop {number} top - The uppermost (lowest) vertical offset for the anchors
+ *   in this bucket — the lowest `top` position value, akin to the top offest of
+ *   a theoretical box drawn around all of the anchor highlights in this bucket
+ * @prop {number} bottom - The bottommost (highest) vertical offset for the
+ *   anchors in this bucket — the highest `top` position value, akin to the
+ *   bottom of a theoretical box drawn around all of the anchor highlights in
+ *   this bucket
  */
 
 /**
- * @typedef Bucket
- * @prop {Anchor[]} anchors - The anchors in this bucket
- * @prop {number} position - The vertical pixel offset where this bucket should
- *                           appear in the bucket bar
+ * @typedef AnchorPosition
+ * @prop {Anchor} anchor
+ * @prop {number} top - The vertical offset, in pixels, of the top of this
+ *   anchor's highlight(s) bounding box
+ * @prop {number} bottom - The vertical offset, in pixels, of the bottom of this
+ *   anchor's highlight(s) bounding box
  */
 
-// FIXME: Temporary duplication of size constants between here and BucketBar
-const BUCKET_SIZE = 16; // Regular bucket size
-const BUCKET_NAV_SIZE = BUCKET_SIZE + 6; // Bucket plus arrow (up/down)
-const BUCKET_TOP_THRESHOLD = 115 + BUCKET_NAV_SIZE; // Toolbar
-
-// TODO!! This is an option in the plugin right now
+// Only anchors with top offsets between `BUCKET_TOP_THRESHOLD` and
+// `BUCKET_BOTTOM_THRESHOLD` are considered "on-screen" and will be bucketed.
+// This is to account for bucket-bar tool buttons (top) and the height of
+// the bottom navigation bucket (bottom)
+const BUCKET_TOP_THRESHOLD = 136.5;
```

Comment:
Why YES, I'm serious, this sub-pixel value makes a difference in label alignment within the upper-navigation bucket. I need to get to the bottom of it.